### PR TITLE
Replace configuration module with xcc.Settings

### PR DESCRIPTION
**Pull request by Mandrenkov**
**Context:**
As part of the migration to the XCC, the configuration module has been replaced by the [xcc.Settings](https://xanadu-cloud-client.readthedocs.io/en/stable/api/xcc.Settings.html) class.

**Description of the Change:**
* Deleted the strawberryfields.configuration module and its corresponding unit tests.
* Added configuration migration notes to the changelog and v0.20.0 migration guide.

**Benefits:**
* Strawberry Fields has less dead code (the only dependents on the strawberryfields.configuration module were removed by #642 and #645).
* There are clear instructions for migrating Strawberry Fields configurations.

**Possible Drawbacks:**
* Some features, such as per-project configuration files, are not (yet) supported by the XCC.

**Related GitHub Issues:**
None.

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -81,75 +81,74 @@
 
   1. Configuring account credentials using:
 
-    * Strawberry Fields v0.19.0
+     * Strawberry Fields v0.19.0
```
Comment (thisac) [inline .github/CHANGELOG.md:84]: Why the added space? Isn't 4 spaces good? 🤔

File: doc/development/migration_guides.rst [hunk 2]
```diff
@@ -26,9 +110,9 @@ each ``sf`` (v0.19.0) command:
    * - ``sf run "foo.xbb"``
      - ``xcc job submit --name "bar" --target "X8_01" --circuit "$(cat foo.xbb)"``
 
-.. note::
+.. warning::
```
Comment (thisac) [inline doc/development/migration_guides.rst:113]: 👍

File: doc/introduction/photonic_hardware.rst [hunk 3]
```diff
@@ -82,14 +82,15 @@ Job objects have the following properties and methods:
 
 Alternatively, if you have your quantum program available as a Blackbird script,
 you can submit the Blackbird script file to be executed remotely using
-the Strawberry Fields command line interface:
+the Xanadu Cloud Client command line interface:
```
Comment (thisac) [inline doc/introduction/photonic_hardware.rst:85]: Good catch! 💯

File: strawberryfields/__init__.py [hunk 4]
```diff
@@ -77,17 +72,18 @@ def about():
         Strawberry Fields: a Python library for continuous-variable quantum circuits.
         Copyright 2018-2020 Xanadu Quantum Technologies Inc.
 
-        Python version:            3.6.10
-        Platform info:             Linux-5.8.18-1-MANJARO-x86_64-with-arch-Manjaro-Linux
+        Python version:            3.9.6
```
Comment (thisac) [inline strawberryfields/__init__.py:75]: Nice to not have Python 3.6 references in the code anymore. 😅

File: strawberryfields/__init__.py [hunk 5]
```diff
@@ -77,17 +72,18 @@ def about():
...
+        Platform info:             Linux-5.10.60.1-microsoft-standard-WSL2-x86_64-with-glibc2.31
```
Comment (thisac) [inline strawberryfields/__init__.py:76]: Are you using WSL? 🪟 🐧

**Review by thisac — APPROVED** [review]
Looks great! Really nice to have this using `XCC` instead. 🥳

File: .github/CHANGELOG.md (hunk 1 above)
Comment (Mandrenkov) [reply .github/CHANGELOG.md:84]: Nope! It turns out we needed one more space for the changelog to render as intended.

File: strawberryfields/__init__.py (hunk 5 above)
Comment (Mandrenkov) [reply strawberryfields/__init__.py:76]: Guilty as charged! 😅

**Pull request merged**